### ISSUE TITLE
Wrong timestamp in database

### ISSUE THREAD

**Issue by azzamsa** [expectocode/telegram-export#96]
Hi.

I just send new message to test the Date value in sqlite. I get `1541441875`.
I tried using GNU date and [python way](https://github.com/expectocode/telegram-export/issues/60#issuecomment-400037351) to translate it. But I always get the wrong result. I also tried online converter which give me the same wrong result.

```
1541441875
Is equivalent to:
11/05/2018 @ 6:17pm (UTC)
```

The corrent time is 

```
My username,, [06.11.18 08:17]
+ foo
```

Why I get the wrong date in database ? Am I missing something ?

Thank you in advance.

**Comment by expectocode** [comment]
@azzamsa I'm guessing this is a time zone issue - are you in UTC - 10? Tl

**Comment by azzamsa** [comment]
This is what i get from terminal

```
username at machinename in ~/.config/telegram-export 
$ date
Tue Nov  6 08:52:01 WIB 2018

username at machinename in ~/.config/telegram-export 
$ date +%s
1541469128

username at machinename in ~/.config/telegram-export 
$ timedatectl status | grep "Time zone"
       Time zone: Asia/Jakarta (WIB, +0700)
```

**Comment by azzamsa** [comment]
@expectocode any solution for this issue ? my dirty fix was add the delta between the right and the wrong one (the original time).

But I hope there will be a fix in upstream

**Comment by expectocode** [comment]
I don't know why this is happening - we've never had an issue with this before, and I can't see a place where it would come from. We get our dates from Telegram :/. Have you tried `date --date='@<timestamp>'` ?

**Comment by azzamsa** [comment]
> date --date='@<timestamp>'

Tried it. Get the same wrong result.

I also tried the [new export feature](https://desktop.telegram.org/changelog#v-1-3-13-27-08-18) from telegram desktop. But the export result shows correct time.

I still prefer to use telegram-export because I can easily automate and parse the result with python. The exported result from default telegam feature is html. Which hard to parse and work with.

Do you have any other idea regarding this wrong timestamp ?, I'm afraid adding deltas between them will result to slightly unpreciese timestamp.

**Comment by Lonami** [comment]
If you use Telethon directly to show the date is it correct in your system?
```python
import datetime
from telethon.sync import TelegramClient
with TelegramClient('name', api_id, api_hash) as client:
    m = client.send_message('me', 'hi')
    m.delete()
    print('Now: ', datetime.datetime.now())
    print('Sent:', m.date)

**Comment by azzamsa** [comment]
I got an error running the code. So I changed little bit according to [this example](https://telethon.readthedocs.io/en/stable/extra/basic/getting-started.html#creating-a-client)

``` python
from telethon import TelegramClient, sync
```

The result is:

``` text
Signed in successfully as User
Now:  2018-11-14 06:11:18.390936
Sent: 2018-11-13 23:11:18+00:00
```

The timestamp error already occurred since the beginning ? from telethon ?

**Comment by Lonami** [comment]
> The timestamp error already occurred since the beginning ? from telethon ?

There is no error from Telethon's side. It has the correct message time (which 23:00:00+07 for your timezone gives the correct 06:00:00). So at least we know we're not carrying a bug from Telethon but rather when saving the timestamp.

**Comment by azzamsa** [comment]
> So at least we know we're not carrying a bug from Telethon but rather when saving the timestamp.

Happy to see telethon author here. Hope we can figure this out.

**Comment by Lonami** [comment]
[`dump_message`](https://github.com/expectocode/telegram-export/blob/c383a3a8eab38b51f218f7bbc261ecab4ffc5d41/telegram_export/dumper.py#L296) [called from this method](https://github.com/expectocode/telegram-export/blob/c383a3a8eab38b51f218f7bbc261ecab4ffc5d41/telegram_export/downloader.py#L127) [without modifying the message anyhow](https://github.com/expectocode/telegram-export/blob/c383a3a8eab38b51f218f7bbc261ecab4ffc5d41/telegram_export/downloader.py#L143) is using the following:
https://github.com/expectocode/telegram-export/blob/c383a3a8eab38b51f218f7bbc261ecab4ffc5d41/telegram_export/dumper.py#L314

So I fail to see where we go wrong.

**Comment by azzamsa** [comment]
I tried to run [this code](https://github.com/expectocode/telegram-export/issues/96#issuecomment-438310184) again.

```
Now:  2018-12-03 10:55:28.131339
Sent: 2018-12-03 03:55:27+00:00
```

I am still curious, I am in +07 UTC, but why the message.date use +00 UTC ?
If this is OK. But why when exporting to sqlite, why telegeram-export doesn't change the sent time to sent+07 which is my correct local time ?

**Comment by Lonami** [comment]
> but why the message.date use +00 UTC

The date is saved in a timezone-agnostic way. That is, local timezones are **not** used, so even if the timezone in which your computer is changes, the saved time will still be correct in UTC+00. If you want to work with the time in your local timezone, post-process the time yourself.

**Comment by azzamsa** [comment]
>  If you want to work with the time in your local timezone, post-process the time yourself.

Yes, this is my workaround for now. But, for you, did you face the same problem ?
are your timestamp in sqlite use UTC+00 or your local timezone ?

**Comment by Lonami** [comment]
There is no problem. `telegram-export` saves all timestamps in UTC+00, which is not wrong.

**Comment by azzamsa** [comment]
So, by mean, we access chat in telegram (which retrieved from server), telegram will post-process the time from UTC+00 to that device local time ?

**Comment by Lonami** [comment]
Yes, all official clients adjust the time to show the local time, which is more user friendly. To save it, however, it's a lot better to use a single time format.

**Comment by azzamsa** [comment]
Oh, this news to me. this is the first time I work with localization.

Thank you for this.

I will add them to +07 manually.

Have a great day.
